core/store.go

```diff
@@ -15,57 +15,57 @@ type WatchEvent struct {
 	Value     *Obj
 }
 
-var (
-	store     map[unsafe.Pointer]*Obj
-	expires   map[*Obj]uint64 // Does not need to be thread-safe as it is only accessed by a single thread.
-	keypool   map[string]unsafe.Pointer
-	WatchList sync.Map // Maps queries to the file descriptors of clients that are watching them.
-)
+type Store struct {
+	store        map[unsafe.Pointer]*Obj
+	expires      map[*Obj]uint64 // Does not need to be thread-safe as it is only accessed by a single thread.
+	keypool      map[string]unsafe.Pointer
+	storeMutex   sync.RWMutex
+	keypoolMutex sync.RWMutex
+	WatchList    sync.Map // Maps queries to the file descriptors of clients that are watching them.
```

Comment:
We are using a mix of sync.Mutex and sync.Map for concurrency control now. How do we plan to refactor this once we move towards no lock?

Reply:
All actions discussed are moved to their separate tasks. Not addressing as part of this change